java/query/query-core/src/main/java/sleeper/query/core/tracker/QueryStatusReportListener.java

```diff
@@ -21,25 +21,79 @@
 
 import java.util.List;
 
+/**
+ * Query status report listener.
```

Comment:
This is just the interface name. We can explain what this is for.

This tracks changes to the status of a query, and is implemented by something that should be notified when the status changes. For example that includes the query tracker, and the web socket for a query that was made through a web socket.